lib/hammer.ex

```diff
@@ -137,38 +91,29 @@ defmodule Hammer do
       {:ok, {1, 2499, 29381612, 1450281014468, 1450281014468}}
 
   """
-  def inspect_bucket(id, scale_ms, limit) do
-    inspect_bucket(:single, id, scale_ms, limit)
-  end
-
-  @spec inspect_bucket(backend :: atom, id :: String.t(), scale_ms :: integer, limit :: integer) ::
+  @spec inspect_bucket(id :: String.t(), scale_ms :: integer, limit :: integer) ::
           {:ok,
            {count :: integer, count_remaining :: integer, ms_to_next_bucket :: integer,
             created_at :: integer | nil, updated_at :: integer | nil}}
           | {:error, reason :: any}
-  @doc """
-  Same as inspect_bucket/3, but allows specifying a backend
-  """
-  def inspect_bucket(backend, id, scale_ms, limit) do
-    {stamp, key} = Utils.stamp_key(id, scale_ms)
-    ms_to_next_bucket = elem(key, 0) * scale_ms + scale_ms - stamp
-
-    case call_backend(backend, :get_bucket, [key]) do
-      {:ok, nil} ->
-        {:ok, {0, limit, ms_to_next_bucket, nil, nil}}
-
-      {:ok, {_, count, created_at, updated_at}} ->
-        count_remaining = if limit > count, do: limit - count, else: 0
-        {:ok, {count, count_remaining, ms_to_next_bucket, created_at, updated_at}}
-
-      {:error, reason} ->
-        {:error, reason}
+  def inspect_bucket(id, scale_ms, limit) do
+    now = System.system_time(:millisecond)
+    now_bucket = div(now, scale_ms)
+    key = {id, now_bucket}
+    ms_to_next_bucket = now_bucket * scale_ms + scale_ms - now
+
+    with {:ok, count} <- @backend.get_bucket(key) do
+      {:ok,
+       {
+         count,
+         max(limit - count, 0),
+         ms_to_next_bucket,
+         _created_at = nil,
+         _updated_at = nil
```

Comment:
I searched for `Hammer.inspect_bucket` on github and it doesn't seem anyone is using `created_at` and `updated_at` values (almost everyone calls this function to get ms_to_next_bucket). So I think `created_at` and `updated_at` can be removed, especially since `created_at` in ETS backend is incorrect after table pruning (cleanup).

Reply:
Yes but we don’t know how people use it in private repo ?

Reply:
I suggest dropping these values and reintroducing them back if someone complains.

Reply:
@ruslandoga I'd suggest adding a deprecation warning first, and then getting rid of them in the version after. That's how Elixir and core libraries do it.